### PR TITLE
Port wasm build to cmake

### PR DESCRIPTION
**Pull request by bakulf**
Wasm compiled using cmake

### CODE REVIEW

File: src/cmake/wasm.cmake [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Linux platform source files
```
Comment (oskirby) [inline src/cmake/wasm.cmake:5]: This comment should be updated

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -34,13 +39,17 @@ elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
...
+    set(MVPN_PLATFORM_NAME "wasm")
 endif()
 
 add_compile_definitions("$<$<CONFIG:Debug>:MVPN_DEBUG>")
 add_compile_definitions("MVPN_$<UPPER_CASE:${MVPN_PLATFORM_NAME}>")
 include(cmake/${MVPN_PLATFORM_NAME}.cmake)
 
-add_subdirectory(crashreporter)
-target_link_libraries(mozillavpn PRIVATE crashreporter)
+if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Emscripten")
```
Comment (oskirby) [inline src/CMakeLists.txt:50]: Would `NOT CROSS_COMPILING` be a better test here? Not that it really matters until we support Android/iOS with CMake.

File: src/addonmanager.cpp [hunk 3]
```diff
@@ -182,8 +182,6 @@ bool AddonManager::validateIndex(const QByteArray& index,
   for (const AddonData& addonData : addons) {
     if (!m_addons.contains(addonData.m_addonId) &&
         validateAndLoad(addonData.m_addonId, addonData.m_sha256)) {
-      Q_ASSERT(m_addons.contains(addonData.m_addonId));
```
Comment (oskirby) [inline src/addonmanager.cpp:185]: This feels unrelated, or has something changed with wasm and the addons?

**Review by oskirby — APPROVED** [review]

File: src/CMakeLists.txt (hunk 2 above)
Comment (bakulf) [reply src/CMakeLists.txt:50]: Well I hope to have crashreport on mobile eventually.

File: src/addonmanager.cpp (hunk 3 above)
Comment (bakulf) [reply src/addonmanager.cpp:185]: you are right. It's unrelated but with a debug build, the wasm client crashes when validating a blocked addon

File: src/CMakeLists.txt (hunk 2 above)
Comment (oskirby) [reply src/CMakeLists.txt:50]: Don't we already get crash reporting through the App/Play stores?

File: taskcluster/scripts/build/wasm.sh [hunk 4]
```diff
@@ -12,18 +12,28 @@ qmake --version
 git submodule init
 git submodule update
 
+wget https://github.com/Kitware/CMake/releases/download/v3.24.0-rc4/cmake-3.24.0-rc4-linux-x86_64.sh
+chmod +x cmake-3.24.0-rc4-linux-x86_64.sh
+mkdir /tmp/cmake
+./cmake-3.24.0-rc4-linux-x86_64.sh --skip-license --prefix=/tmp/cmake
```
Comment (strseb) [inline taskcluster/scripts/build/wasm.sh:18]: NIT: I think this should go here :)
 https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/taskcluster/docker/wasm/Dockerfile

File: src/CMakeLists.txt (hunk 2 above)
Comment (bakulf) [reply src/CMakeLists.txt:50]: not via crashpad
